src/main/java/org/folio/util/AsyncProcessingContext.java

```diff
@@ -0,0 +1,29 @@
+package org.folio.util;
+
+import static java.lang.String.format;
+
+import java.lang.invoke.MethodHandles;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+public abstract class AsyncProcessingContext {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  protected abstract String getName();
+
+  public void logFailedValidationError(String failedStep, String additionalMessage) {
```

Comment:
Why is it context's responsibility to do the logging?

Reply:
This logging is about context validation errors. `UserSummaryService` and `PatronBlocksService` both use such contexts so it doesn't make sense to duplicate the logging logic.